packages/jest/src/generators/convert-to-inferred/convert-to-inferred.ts

```diff
@@ -0,0 +1,210 @@
+import type { Config } from '@jest/types';
+import {
+  createProjectGraphAsync,
+  formatFiles,
+  workspaceRoot,
+  type TargetConfiguration,
+  type Tree,
+} from '@nx/devkit';
+import { migrateExecutorToPlugin } from '@nx/devkit/src/generators/plugin-migrations/executor-to-plugin-migrator';
+import {
+  processTargetOutputs,
+  toProjectRelativePath,
+} from '@nx/devkit/src/generators/plugin-migrations/plugin-migration-utils';
+import { readConfig } from 'jest-config';
+import { join, normalize } from 'node:path/posix';
+import { createNodes } from '../../plugins/plugin';
+
+interface Schema {
+  project?: string;
+  skipFormat?: boolean;
+}
+
+export async function convertToInferred(tree: Tree, options: Schema) {
+  const projectGraph = await createProjectGraphAsync();
+  const migratedProjectsModern = await migrateExecutorToPlugin(
+    tree,
+    projectGraph,
+    '@nx/jest:jest',
+    '@nx/jest/plugin',
+    (targetName) => ({ targetName }),
+    postTargetTransformer,
+    createNodes,
+    options.project
+  );
+
+  const migratedProjectsLegacy = await migrateExecutorToPlugin(
+    tree,
+    projectGraph,
+    '@nrwl/jest:jest',
+    '@nx/jest/plugin',
+    (targetName) => ({ targetName }),
+    postTargetTransformer,
+    createNodes,
+    options.project
+  );
+
+  const migratedProjects =
+    migratedProjectsModern.size + migratedProjectsLegacy.size;
+
+  if (migratedProjects === 0) {
+    throw new Error('Could not find any targets to migrate.');
+  }
+
+  if (!options.skipFormat) {
+    await formatFiles(tree);
+  }
+}
+
+async function postTargetTransformer(
+  target: TargetConfiguration,
+  tree: Tree,
+  projectDetails: { projectName: string; root: string },
+  inferredTarget: TargetConfiguration
+): Promise<TargetConfiguration> {
+  if (target.options) {
+    const jestConfigPath = target.options.jestConfig;
```

Comment:
Is there any chance this can be undefined and the jest executor still works?
I ask because you use this value here: https://github.com/nrwl/nx/pull/26259/files#diff-c440c2d41bff855c8035e19fe4948557afa0785f9508f9dfba1ce53cb7856442R119

Reply:
That's a required option, so it _should_ always be defined. The executor doesn't work without it. That said, I guess there's nothing stopping folks from having an unused, invalid target without it. What are we doing in those cases?